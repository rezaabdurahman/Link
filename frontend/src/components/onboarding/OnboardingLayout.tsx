// OnboardingLayout - Layout component for onboarding flow
// Provides consistent styling, header, and navigation for all onboarding steps

import React from 'react';
import { Link } from 'react-router-dom';
import { ArrowLeft } from 'lucide-react';
import { useAuth } from '../../contexts/AuthContext';
import { useOnboarding } from '../../contexts/OnboardingContext';

interface OnboardingLayoutProps {
  children: React.ReactNode;
  showBackButton?: boolean;
  showSkipButton?: boolean;
  onBack?: () => void;
  onSkip?: () => void;
}

const OnboardingLayout: React.FC<OnboardingLayoutProps> = ({
  children,
  showBackButton = true,
  onBack,
}): JSX.Element => {
  const { user } = useAuth();
  const { 
    canGoPrevious, 
    goToPreviousStep,
    isLoading 
  } = useOnboarding();

  const handleBack = async (): Promise<void> => {
    if (onBack) {
      onBack();
    } else if (canGoPrevious) {
      try {
        await goToPreviousStep();
      } catch (error) {
        console.error('Failed to go to previous step:', error);
      }
    }
  };


  return (
    <div className="min-h-screen bg-surface-dark">
      {/* Header */}
      <header className="ios-card border-b border-gray-200/50">
        <div className="max-w-7xl mx-auto px-4 sm:px-6 lg:px-8">
          <div className="flex justify-between items-center py-6">
            {/* Left side - Back button */}
            <div className="flex items-center space-x-4">
              {showBackButton && canGoPrevious && (
                <button
                  onClick={handleBack}
                  disabled={isLoading}
                  className="ios-button-secondary p-2 text-text-secondary hover:text-text-primary transition-colors disabled:opacity-50 disabled:cursor-not-allowed"
                  aria-label="Go back"
                >
                  <ArrowLeft className="w-5 h-5" />
                </button>
              )}
              
              {/* Logo */}
              <Link to="/discovery" className="flex items-center">
                <div className="w-8 h-8 bg-aqua rounded-lg flex items-center justify-center">
                  <span className="text-white font-bold text-sm">L</span>
                </div>
                <span className="ml-2 text-xl font-semibold text-primary">Link</span>
              </Link>
            </div>

            {/* Center - User greeting */}
            <div className="hidden sm:block">
              {user && (
                <p className="text-sm text-secondary">
                  Welcome, {user.first_name}!
                </p>
              )}
            </div>

            {/* Right side - Empty for now */}
            <div className="flex items-center">
              {/* No skip button - onboarding is required */}
            </div>
          </div>
        </div>
      </header>

      {/* Main Content */}
      <main className="flex-1">
        <div className="max-w-7xl mx-auto px-4 sm:px-6 lg:px-8 py-6">
          {children}
        </div>
      </main>

      {/* Footer */}
      <footer className="bg-surface-card border-t border-gray-200 mt-auto">
        <div className="max-w-7xl mx-auto px-4 sm:px-6 lg:px-8 py-4">
          <div className="flex justify-center items-center space-x-6 text-sm text-muted">
            <Link 
              to="/help" 
              className="hover:text-primary transition-colors"
            >
              Help
            </Link>
            <Link 
              to="/privacy" 
              className="hover:text-primary transition-colors"
            >
              Privacy
            </Link>
            <Link 
              to="/terms" 
              className="hover:text-primary transition-colors"
            >
              Terms
            </Link>
          </div>
        </div>
      </footer>
    </div>
  );
};

export default OnboardingLayout;
