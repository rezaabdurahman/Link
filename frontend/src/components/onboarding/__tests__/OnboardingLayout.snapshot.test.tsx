import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import '@testing-library/jest-dom';

// Mock the contexts to avoid import issues
const mockAuthContext = {
  user: {
    id: 'test-user-id',
    first_name: 'John',
    last_name: 'Doe',
    email: 'john.doe@example.com',
    bio: 'Test bio',
  },
  isAuthenticated: true,
  isLoading: false,
};

const mockOnboardingContext = {
  currentStepData: {},
  setStepData: jest.fn(),
  goToNextStep: jest.fn().mockResolvedValue(undefined),
  goToPreviousStep: jest.fn().mockResolvedValue(undefined),
  skipCurrentStep: jest.fn().mockResolvedValue(undefined),
  canGoPrevious: true,
  canGoNext: true,
  updateUserProfile: jest.fn().mockResolvedValue(undefined),
  completeOnboardingFlow: jest.fn().mockResolvedValue(undefined),
  isLoading: false,
  error: null,
  clearError: jest.fn(),
};

jest.mock('../../../contexts/AuthContext', () => ({
  useAuth: () => mockAuthContext,
}));

jest.mock('../../../contexts/OnboardingContext', () => ({
  useOnboarding: () => mockOnboardingContext,
}));

// Mock framer-motion
jest.mock('framer-motion', () => ({
  motion: {
    div: ({ children, className, ...props }: any) => (
      <div className={className} {...props}>
        {children}
      </div>
    ),
  },
  AnimatePresence: ({ children }: any) => <div>{children}</div>,
}));

// Import the component after mocking
import OnboardingLayout from '../OnboardingLayout';

describe('OnboardingLayout Snapshots', () => {
  const renderWithRouter = (ui: React.ReactElement) => {
    return render(
      <BrowserRouter>
        {ui}
      </BrowserRouter>
    );
  };

  it('should match snapshot with default props', () => {
    const { container } = renderWithRouter(
      <OnboardingLayout>
        <div>Test Content</div>
      </OnboardingLayout>
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should match snapshot with back button visible', () => {
    // Update mock to show back button
    mockOnboardingContext.canGoPrevious = true;

    const { container } = renderWithRouter(
      <OnboardingLayout showBackButton={true}>
        <div>Test Content with Back Button</div>
      </OnboardingLayout>
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should match snapshot with user greeting', () => {
    const { container } = renderWithRouter(
      <OnboardingLayout>
        <div>Personalized Content</div>
      </OnboardingLayout>
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should match snapshot in loading state', () => {
    // Update mock to loading state
    mockOnboardingContext.isLoading = true;
    mockOnboardingContext.canGoPrevious = true;

    const { container } = renderWithRouter(
      <OnboardingLayout showBackButton={true}>
        <div>Loading State Content</div>
      </OnboardingLayout>
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
