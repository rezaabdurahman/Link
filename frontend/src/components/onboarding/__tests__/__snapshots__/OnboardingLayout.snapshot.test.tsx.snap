// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OnboardingLayout Snapshots should match snapshot in loading state 1`] = `
<div
  class="min-h-screen bg-surface-dark"
>
  <header
    class="ios-card border-b border-gray-200/50"
  >
    <div
      class="max-w-7xl mx-auto px-4 sm:px-6 lg:px-8"
    >
      <div
        class="flex justify-between items-center py-6"
      >
        <div
          class="flex items-center space-x-4"
        >
          <button
            aria-label="Go back"
            class="ios-button-secondary p-2 text-text-secondary hover:text-text-primary transition-colors disabled:opacity-50 disabled:cursor-not-allowed"
            disabled=""
          >
            <svg
              class="lucide lucide-arrow-left w-5 h-5"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m12 19-7-7 7-7"
              />
              <path
                d="M19 12H5"
              />
            </svg>
          </button>
          <a
            class="flex items-center"
            href="/discovery"
          >
            <div
              class="w-8 h-8 bg-aqua rounded-lg flex items-center justify-center"
            >
              <span
                class="text-white font-bold text-sm"
              >
                L
              </span>
            </div>
            <span
              class="ml-2 text-xl font-semibold text-primary"
            >
              Link
            </span>
          </a>
        </div>
        <div
          class="hidden sm:block"
        >
          <p
            class="text-sm text-secondary"
          >
            Welcome, 
            John
            !
          </p>
        </div>
        <div
          class="flex items-center"
        />
      </div>
    </div>
  </header>
  <main
    class="flex-1"
  >
    <div
      class="max-w-7xl mx-auto px-4 sm:px-6 lg:px-8 py-6"
    >
      <div>
        Loading State Content
      </div>
    </div>
  </main>
  <footer
    class="bg-surface-card border-t border-gray-200 mt-auto"
  >
    <div
      class="max-w-7xl mx-auto px-4 sm:px-6 lg:px-8 py-4"
    >
      <div
        class="flex justify-center items-center space-x-6 text-sm text-muted"
      >
        <a
          class="hover:text-primary transition-colors"
          href="/help"
        >
          Help
        </a>
        <a
          class="hover:text-primary transition-colors"
          href="/privacy"
        >
          Privacy
        </a>
        <a
          class="hover:text-primary transition-colors"
          href="/terms"
        >
          Terms
        </a>
      </div>
    </div>
  </footer>
</div>
`;

exports[`OnboardingLayout Snapshots should match snapshot with back button visible 1`] = `
<div
  class="min-h-screen bg-surface-dark"
>
  <header
    class="ios-card border-b border-gray-200/50"
  >
    <div
      class="max-w-7xl mx-auto px-4 sm:px-6 lg:px-8"
    >
      <div
        class="flex justify-between items-center py-6"
      >
        <div
          class="flex items-center space-x-4"
        >
          <button
            aria-label="Go back"
            class="ios-button-secondary p-2 text-text-secondary hover:text-text-primary transition-colors disabled:opacity-50 disabled:cursor-not-allowed"
          >
            <svg
              class="lucide lucide-arrow-left w-5 h-5"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m12 19-7-7 7-7"
              />
              <path
                d="M19 12H5"
              />
            </svg>
          </button>
          <a
            class="flex items-center"
            href="/discovery"
          >
            <div
              class="w-8 h-8 bg-aqua rounded-lg flex items-center justify-center"
            >
              <span
                class="text-white font-bold text-sm"
              >
                L
              </span>
            </div>
            <span
              class="ml-2 text-xl font-semibold text-primary"
            >
              Link
            </span>
          </a>
        </div>
        <div
          class="hidden sm:block"
        >
          <p
            class="text-sm text-secondary"
          >
            Welcome, 
            John
            !
          </p>
        </div>
        <div
          class="flex items-center"
        />
      </div>
    </div>
  </header>
  <main
    class="flex-1"
  >
    <div
      class="max-w-7xl mx-auto px-4 sm:px-6 lg:px-8 py-6"
    >
      <div>
        Test Content with Back Button
      </div>
    </div>
  </main>
  <footer
    class="bg-surface-card border-t border-gray-200 mt-auto"
  >
    <div
      class="max-w-7xl mx-auto px-4 sm:px-6 lg:px-8 py-4"
    >
      <div
        class="flex justify-center items-center space-x-6 text-sm text-muted"
      >
        <a
          class="hover:text-primary transition-colors"
          href="/help"
        >
          Help
        </a>
        <a
          class="hover:text-primary transition-colors"
          href="/privacy"
        >
          Privacy
        </a>
        <a
          class="hover:text-primary transition-colors"
          href="/terms"
        >
          Terms
        </a>
      </div>
    </div>
  </footer>
</div>
`;

exports[`OnboardingLayout Snapshots should match snapshot with default props 1`] = `
<div
  class="min-h-screen bg-surface-dark"
>
  <header
    class="ios-card border-b border-gray-200/50"
  >
    <div
      class="max-w-7xl mx-auto px-4 sm:px-6 lg:px-8"
    >
      <div
        class="flex justify-between items-center py-6"
      >
        <div
          class="flex items-center space-x-4"
        >
          <button
            aria-label="Go back"
            class="ios-button-secondary p-2 text-text-secondary hover:text-text-primary transition-colors disabled:opacity-50 disabled:cursor-not-allowed"
          >
            <svg
              class="lucide lucide-arrow-left w-5 h-5"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m12 19-7-7 7-7"
              />
              <path
                d="M19 12H5"
              />
            </svg>
          </button>
          <a
            class="flex items-center"
            href="/discovery"
          >
            <div
              class="w-8 h-8 bg-aqua rounded-lg flex items-center justify-center"
            >
              <span
                class="text-white font-bold text-sm"
              >
                L
              </span>
            </div>
            <span
              class="ml-2 text-xl font-semibold text-primary"
            >
              Link
            </span>
          </a>
        </div>
        <div
          class="hidden sm:block"
        >
          <p
            class="text-sm text-secondary"
          >
            Welcome, 
            John
            !
          </p>
        </div>
        <div
          class="flex items-center"
        />
      </div>
    </div>
  </header>
  <main
    class="flex-1"
  >
    <div
      class="max-w-7xl mx-auto px-4 sm:px-6 lg:px-8 py-6"
    >
      <div>
        Test Content
      </div>
    </div>
  </main>
  <footer
    class="bg-surface-card border-t border-gray-200 mt-auto"
  >
    <div
      class="max-w-7xl mx-auto px-4 sm:px-6 lg:px-8 py-4"
    >
      <div
        class="flex justify-center items-center space-x-6 text-sm text-muted"
      >
        <a
          class="hover:text-primary transition-colors"
          href="/help"
        >
          Help
        </a>
        <a
          class="hover:text-primary transition-colors"
          href="/privacy"
        >
          Privacy
        </a>
        <a
          class="hover:text-primary transition-colors"
          href="/terms"
        >
          Terms
        </a>
      </div>
    </div>
  </footer>
</div>
`;

exports[`OnboardingLayout Snapshots should match snapshot with user greeting 1`] = `
<div
  class="min-h-screen bg-surface-dark"
>
  <header
    class="ios-card border-b border-gray-200/50"
  >
    <div
      class="max-w-7xl mx-auto px-4 sm:px-6 lg:px-8"
    >
      <div
        class="flex justify-between items-center py-6"
      >
        <div
          class="flex items-center space-x-4"
        >
          <button
            aria-label="Go back"
            class="ios-button-secondary p-2 text-text-secondary hover:text-text-primary transition-colors disabled:opacity-50 disabled:cursor-not-allowed"
          >
            <svg
              class="lucide lucide-arrow-left w-5 h-5"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m12 19-7-7 7-7"
              />
              <path
                d="M19 12H5"
              />
            </svg>
          </button>
          <a
            class="flex items-center"
            href="/discovery"
          >
            <div
              class="w-8 h-8 bg-aqua rounded-lg flex items-center justify-center"
            >
              <span
                class="text-white font-bold text-sm"
              >
                L
              </span>
            </div>
            <span
              class="ml-2 text-xl font-semibold text-primary"
            >
              Link
            </span>
          </a>
        </div>
        <div
          class="hidden sm:block"
        >
          <p
            class="text-sm text-secondary"
          >
            Welcome, 
            John
            !
          </p>
        </div>
        <div
          class="flex items-center"
        />
      </div>
    </div>
  </header>
  <main
    class="flex-1"
  >
    <div
      class="max-w-7xl mx-auto px-4 sm:px-6 lg:px-8 py-6"
    >
      <div>
        Personalized Content
      </div>
    </div>
  </main>
  <footer
    class="bg-surface-card border-t border-gray-200 mt-auto"
  >
    <div
      class="max-w-7xl mx-auto px-4 sm:px-6 lg:px-8 py-4"
    >
      <div
        class="flex justify-center items-center space-x-6 text-sm text-muted"
      >
        <a
          class="hover:text-primary transition-colors"
          href="/help"
        >
          Help
        </a>
        <a
          class="hover:text-primary transition-colors"
          href="/privacy"
        >
          Privacy
        </a>
        <a
          class="hover:text-primary transition-colors"
          href="/terms"
        >
          Terms
        </a>
      </div>
    </div>
  </footer>
</div>
`;
